lib/cookie/cookie.ts

```diff
@@ -401,23 +426,101 @@ const cookieDefaults = {
   sameSite: undefined,
 } as const satisfies Required<CreateCookieOptions>
 
+/**
+ * An HTTP cookie (web cookie, browser cookie) is a small piece of data that a server sends to a user's web browser.
+ * It is defined in {@link https://www.rfc-editor.org/rfc/rfc6265.html | RFC6265}.
+ * @public
+ */
 export class Cookie {
```

Comment:
```suggestion
export class Cookie implements Required<CreateCookieOptions> {
```
`CreateCookieOptions` previously was computed from `Cookie`. Now that they are both declared independently, we could use `implements` to guard against accidentally diverging definitions.

Reply:
I'm not sure this is necessary. It would catch a new field added to `CreateCookieOptions` but it wouldn't catch a new field added to `Cookie` so these two can still get out of sync. Also, it stands out to me as a weird type signature if we're implementing a class against it's creation options. 

I believe the risk of these definitions getting out of sync is low since any code to support a new or renamed field would inevitably force us to modify both `Cookie` and `CreateCookieOptions` as well as other definitions.

But now that we have a solid foundation for TypeScript in this library I think we should look at revising our types to tackle some of these issues in our next major release.